assets/scripts/segments/drag_and_drop.js

```diff
@@ -476,7 +505,7 @@ let oldDraggingState = store.getState().ui.draggingState
 // This prevents a constant dispatch of the updateDraggingState action which causes the
 // dragging of the segment to be laggy and choppy.
 
-function updateIfDraggingStateChanged (segmentBeforeEl, segmentAfterEl, draggedItem) {
+function updateIfDraggingStateChanged (segmentBeforeEl, segmentAfterEl, draggedItem, draggedItemType) {
   let changed = false
 
   if (oldDraggingState) {
```

Comment:
For the first drag and drop action that occurs on the street, if you drop the segment right away, it causes the segment to add itself to the front of the street. I'm thinking it has to do with the draggingState no longer being `null` when nothing is being dragged.

Reply:
Good catch. I think it's related to the above -- I hadn't realized there was a lot of code that checked to see if certain values were `undefined`.

We usually have the pattern where if some property (on an object) should sometimes be defined but other times it's not actually equal to anything (e.g. values in state, store, or otherwise) we explicitly initialize it and set its value to `null`. The thing I was trying to solve for was so that one does not need to verify that the `draggingState` exists before trying to read a property off of it. Then, instead of checking whether a property is `undefined` we should just check to see if it is `null` or not.

Reply:
I tried to go all the way with this, and made it worse. So I'm going to revert that change part-way and I think the issues are resolved.